### PR TITLE
integrate transformer app

### PR DESCRIPTION
**Pull request by rshewitt**
# Pull Request

Related to [#4565](https://github.com/GSA/data.gov/issues/4565)

## About
- integrate MDTranslator/transformation app
- add transform tests and fixtures
- update harvest source model with enumeration for schema_type and source_type
  - update all references/instances of this model with appropriate values 

## PR TASKS

- [x] The actual code changes.
- [x] Tests written and passed.
- [ ] Any changes to docs?

### CODE REVIEW

File: app/forms.py [hunk 1]
```diff
@@ -36,10 +36,12 @@ class HarvestSourceForm(FlaskForm):
         validators=[DataRequired()],
     )
     schema_type = SelectField(
-        "Schema Type", choices=["strict", "other"], validators=[DataRequired()]
+        "Schema Type",
+        choices=["iso19115_1", "iso19115_2", "csdgm", "dcatus"],
```
Comment (jbrown-xentity) [inline app/forms.py]: As we know that DCAT-US 3.0 is coming, we should really specify the version in dcatus, something like dcatus1.1.

File: harvester/harvest.py [hunk 2]
```diff
@@ -292,14 +314,16 @@ def synchronize_records(self) -> None:
                     # no longer setting action in compare so setting it here...
                     record.action = action
 
+                    if self.schema_type != "dcatus":
+                        record.transform()
                     record.validate()
```
Comment (jbrown-xentity) [inline harvester/harvest.py:319]: The validation of this won't work with federal validation as they won't have bureau and program code. We'll need to pick up this ticket for this to fully function: https://github.com/GSA/data.gov/issues/4870

**Review by jbrown-xentity — DISMISSED** [review]
Looking good! Couple of comments...

File: app/forms.py (hunk 1 above)
Comment (rshewitt) [reply app/forms.py]: added

**Review by jbrown-xentity — APPROVED** [review]

**Pull request merged**